packages/ra-core/src/controller/list/useListParams.ts

```diff
@@ -111,7 +119,10 @@ export const useListParams = ({
         () =>
             getQuery({
                 queryFromLocation,
-                params: disableSyncWithLocation ? localParams : params,
+                params:
+                    disableSyncWithLocation || disableSyncWithStore
```

Comment:
I think you must also dd the check for `disableSyncWithStore` in `changeParams` L146, and anywhere else the `setParams` may be called.

Reply:
`changeParams` does not call `setParams`